### PR TITLE
update github.com/pkg/errors to go native errors pkg in staging

### PR DESCRIPTION
**Pull request by youhonglian**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes # Part of #103043

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by pacoxu — COMMENTED** [review]
/area code-organization
/kind bug
/ok-to-test

**Comment by youhonglian** [comment]
I updated errors.New(), is that right?

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go [hunk 1]
```diff
@@ -658,17 +656,17 @@ func (meal *cfgMeal) finishQueueSetReconfigsLocked() {
 // object is malformed in a way that is a problem for this package.
 func queueSetCompleterForPL(qsf fq.QueueSetFactory, queues fq.QueueSet, pl *flowcontrol.PriorityLevelConfiguration, requestWaitLimit time.Duration, intPair metrics.TimedObserverPair) (fq.QueueSetCompleter, error) {
 	if (pl.Spec.Type == flowcontrol.PriorityLevelEnablementExempt) != (pl.Spec.Limited == nil) {
-		return nil, errors.New("broken union structure at the top")
+		return nil, fmt.Errorf("broken union structure at the top")
```
Comment (lavalamp) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go:659]: I thought `errors.New` was recommended when there's no arguments? (but referring to the std lib `"errors"`, not this other package)

Comment (lavalamp) [reply staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go:659]: Occurs twice below, too. Otherwise the change looks fine.

Comment (youhonglian) [reply staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go]: > I thought `errors.New` was recommended when there's no arguments? (but referring to the std lib `"errors"`, not this other package)

I changed to the lib `errors` and changed `errors.New` back, thanks for reminding

**Comment by lavalamp** [comment]
/lgtm
/approve

thanks!

**Comment by MikeSpreitzer** [comment]
@youhonglian: thanks for taking care of this.

**Pull request merged**